modules/update-using-custom-machine-config-pools-inheritance.adoc

```diff
@@ -0,0 +1,224 @@
+[source,terminal]
+----
+machineconfigpool.machineconfiguration.openshift.io/worker-perf-canary created
+----
+
+. Check if the `MachineConfig` is inherited in `worker-perf-canary`. 
+.. Verify that no MCP is degraded by running the following command:
++
+[source,terminal]
+----
+$ oc get mcp
+----
++
+.Example output
+[source,terminal]
+----
+NAME                  CONFIG                                                          UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
+master                rendered-master-2bf1379b39e22bae858ea1a3ff54b2ac                True      False      False      3              3                   3                     0                      5d16h
+worker                rendered-worker-b9576d51e030413cfab12eb5b9841f34                True      False      False      0              0                   0                     0                      5d16h
+worker-perf          rendered-worker-perf-b98a1f62485fa702c4329d17d9364f6a          True      False      False      2              2                   2                     0                      56m
+worker-perf-canary   rendered-worker-perf-canary-b98a1f62485fa702c4329d17d9364f6a   True      False      False      1              1                   1                     0                      44m
+----
+
+.. Verify that the machines are inherited from `worker-perf` into `worker-perf-canary`.
++
+[source,terminal]
+----
+$ oc get nodes
+----
++
+.Example output
+[source,terminal]
+----
+NAME       STATUS   ROLES                        AGE     VERSION
+...
+worker-a   Ready    worker,worker-perf-canary   5d15h   v1.27.13+e709aa5
+worker-b   Ready    worker,worker-perf          5d15h   v1.27.13+e709aa5
+worker-c   Ready    worker,worker-perf          5d15h   v1.27.13+e709aa5
+----
+
+.. Verify that `kdump` service is enabled on `worker-a` by running the following command:
++
+[source,terminal]
+----
+$ systemctl status kdump.service
+----
++
+.Example output
+[source,terminal]
+----
+NAME       STATUS   ROLES                        AGE     VERSION
+...
+kdump.service - Crash recovery kernel arming
+     Loaded: loaded (/usr/lib/systemd/system/kdump.service; enabled; preset: disabled)
+     Active: active (exited) since Tue 2024-09-03 12:44:43 UTC; 10s ago
+    Process: 4151139 ExecStart=/usr/bin/kdumpctl start (code=exited, status=0/SUCCESS)
+   Main PID: 4151139 (code=exited, status=0/SUCCESS)
+----
+
+.. Verify that the MCP has updated the `crashkernel` by running the following command:
++
+[source,terminal]
+----
+$ cat /proc/cmdline
+----
++
+The output should include the updated `crashekernel` value, for example:
++
+.Example output
+[source,terminal]
+----
+crashkernel=512M
+----
+
+. (Optional) If you are satisfied with the upgrade, you can return `worker-a` to `worker-perf` by running the following command:
++
+[source,terminal]
+----
+$ oc label node worker-a node-role.kubernetes.io/worker-perf=''
```

Comment:
To return the node from worker-perf-canary MCP to worker-perf again, apart from adding the worker-perf label we need to remove the worker-perf-canary label.

If we don't remove the worker-perf-canary label, the node will be long to 2 MCPs at the same time and the controller will report an error because it is forbidden.

Reply:
Updated the step. Thanks